### DISCUSSION TITLE
Error relevant to parallelism

### DISCUSSION THREAD

**Discussion by jaejae9804** [deepmodeling/deepmd-kit#2734] [Q&A]
After installing the Lammps built-in mode and compiling it with DeepMDKIT, I got this execution file lmp_cmse_deepmd. So right now I can execute LAMMPS using deepmd potential by using the command "lmp_cmse_deepmd -i input.lammps". But here's what I've encountered.
 

LAMMPS (2 Aug 2023)
OMP_NUM_THREADS environment is not set. Defaulting to 1 thread. (../comm.cpp:98)
  using 1 OpenMP thread(s) per MPI task
Reading data file ...
  orthogonal box = (0 0 0) to (100 100 100)
  1 by 1 by 1 MPI processor grid
  reading atoms ...
  74978 atoms
  read_data CPU = 0.336 seconds
DeePMD-kit WARNING: Environmental variable TF_INTRA_OP_PARALLELISM_THREADS is not set. Tune TF_INTRA_OP_PARALLELISM_THREADS for the best performance. See https://deepmd.rtfd.io/parallelism/ for more information.
DeePMD-kit WARNING: Environmental variable TF_INTER_OP_PARALLELISM_THREADS is not set. Tune TF_INTER_OP_PARALLELISM_THREADS for the best performance. See https://deepmd.rtfd.io/parallelism/ for more information.
DeePMD-kit WARNING: Environmental variable OMP_NUM_THREADS is not set. Tune OMP_NUM_THREADS for the best performance. See https://deepmd.rtfd.io/parallelism/ for more information.
Summary of lammps deepmd module ...
  >>> Info of deepmd-kit:
  installed to:       
  source:             v2.2.2-85-g53a1078
  source branch:       master
  source commit:      53a1078
  source commit at:   2023-08-08 11:52:54 +0800
  surpport model ver.:1.1 
  build variant:      cpu
  build with tf inc:  /home/mesiya9804/anaconda3/envs/tf24/lib/python3.8/site-packages/tensorflow/include;/home/mesiya9804/anaconda3/envs/tf24/lib/python3.8/site-packages/tensorflow/include
  build with tf lib:  /home/mesiya9804/anaconda3/envs/tf24/lib/python3.8/site-packages/tensorflow/python/_pywrap_tensorflow_internal.so
  set tf intra_op_parallelism_threads: 0
  set tf inter_op_parallelism_threads: 0
  >>> Info of lammps module:
  use deepmd-kit at:  
  source:             v2.2.2-85-g53a1078
  source branch:      master
  source commit:      53a1078
  source commit at:   2023-08-08 11:52:54 +0800
  build float prec:   double
  build with tf inc:  /home/mesiya9804/anaconda3/envs/tf24/lib/python3.8/site-packages/tensorflow/include;/home/mesiya9804/anaconda3/envs/tf24/lib/python3.8/site-packages/tensorflow/include
  build with tf lib:  /home/mesiya9804/anaconda3/envs/tf24/lib/python3.8/site-packages/tensorflow/python/_pywrap_tensorflow_internal.so
DeePMD-kit WARNING: Environmental variable TF_INTRA_OP_PARALLELISM_THREADS is not set. Tune TF_INTRA_OP_PARALLELISM_THREADS for the best performance. See https://deepmd.rtfd.io/parallelism/ for more information.
DeePMD-kit WARNING: Environmental variable TF_INTER_OP_PARALLELISM_THREADS is not set. Tune TF_INTER_OP_PARALLELISM_THREADS for the best performance. See https://deepmd.rtfd.io/parallelism/ for more information.
DeePMD-kit WARNING: Environmental variable OMP_NUM_THREADS is not set. Tune OMP_NUM_THREADS for the best performance. See https://deepmd.rtfd.io/parallelism/ for more information.
2023-08-10 16:05:29.631132: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX512F
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2023-08-10 16:05:29.674550: I tensorflow/compiler/mlir/mlir_graph_optimization_pass.cc:196] None of the MLIR optimization passes are enabled (registered 0 passes)
Segmentation fault (core dumped)

During the installing process, I've used the following commands to install the relevant programs
1. pip install --upgrade pip        => for upgrading pip
2. conda create -n tf24 python=3.8    => to create virtual env named tf24 with python version of 3.8
3. pip install tensorflow==2.4            => to install tensorflow wersion of 2.4

Im not sure if the three components are related to the error..
Can I have some advice about the error?
Any help will be greatle appreciated.

**Comment by njzjz** [comment]
Could you share the input files to help us reproduce the problem? Thanks.
